Configure

```diff
@@ -1723,21 +1730,19 @@ while (<IN>)
 	s/^PLATFORM=.*$/PLATFORM=$target/;
 	s/^OPTIONS=.*$/OPTIONS=$options/;
 	s/^CONFIGURE_ARGS=.*$/CONFIGURE_ARGS=$argvstring/;
+	s/^MAKEDEPPROG=.*$/MAKEDEPPROG= $makedepprog/ if defined($makedepprog);
+	s/^CC=.*$/CC= $cc/;
+	s/^AR=\s*ar/AR= $ar/;
+	s/^RANLIB=.*/RANLIB= $ranlib/;
+	s/^RC=.*/RC= $windres/;
 	if ($cross_compile_prefix)
 		{
-		s/^CC=.*$/CROSS_COMPILE= $cross_compile_prefix\nCC= \$\(CROSS_COMPILE\)$cc/;
+		s/^CC=\s*/CROSS_COMPILE= $cross_compile_prefix\nCC= \$\(CROSS_COMPILE\)/;
 		s/^AR=\s*/AR= \$\(CROSS_COMPILE\)/;
 		s/^NM=\s*/NM= \$\(CROSS_COMPILE\)/;
 		s/^RANLIB=\s*/RANLIB= \$\(CROSS_COMPILE\)/;
 		s/^RC=\s*/RC= \$\(CROSS_COMPILE\)/;
-		s/^MAKEDEPPROG=.*$/MAKEDEPPROG= \$\(CROSS_COMPILE\)$cc/ if $predefined{__GNUC__} >= 3;
-		}
-	else	{
-		s/^CC=.*$/CC= $cc/;
-		s/^AR=\s*ar/AR= $ar/;
-		s/^RANLIB=.*/RANLIB= $ranlib/;
-		s/^RC=.*/RC= $windres/;
-		s/^MAKEDEPPROG=.*$/MAKEDEPPROG= $cc/ if $predefined{__GNUC__} >= 3;
+		s/^MAKEDEPPROG=\s*/MAKEDEPPROG= \$\(CROSS_COMPILE\)/;
```

Comment:
Original possible outcomes were makedepend or $(CROSS_COMPILE)$cc. Now $(CROSS_COMPILE)makedepend becomes a possibility, and I don't see that it's viable possibility. Then originally the only thing that could happen when cross-compile prefix is defined was that it's injected into AR=ar, NM=nm, RANLIB=ranlib, RC=windres. In other words you had no option to replace ar, nm, ranlib and windres. Now it becomes possibility. One can argue that it's meaningful thing to do, but one can as well argue that long-term support branch is not right place for that kind of modifications. Or in other words modifications should preserve original behaviour.

Reply:
I hear you; I thought it went a bit far, too, but decided to try it. As you point out, might very well be "correct" behavior, but too much for LTS.